packages/flutter_tools/lib/src/aot.dart

```diff
@@ -82,6 +83,9 @@ class AotBuilder {
       buildDir: FlutterProject.current().dartTool.childDirectory('flutter_build'),
       cacheDir: null,
       flutterRootDir: globals.fs.directory(Cache.flutterRoot),
+      engineVersion: globals.artifacts is LocalEngineArtifacts
```

Comment:
The `is LocalEngineArtifacts` pattern could cause confusion if `globals.artifacts` is an `OverrideArtifacts` that delegates to a `LocalEngineArtifacts`, for example. Another question would be which path should the Artifacts implementation in g3 take?

Reply:
Since the google3 build doesn't use assemble at all, it doesn't really matter. I updated this pattern so that instead of doing an `is` check there is an `isLocalEngine` getter.

`OverrideArtifacts` delegates to the parent in this case